test/e2e/autoscale_test.go

```diff
@@ -84,15 +95,26 @@ func runAutoscaleUpCountPods(t *testing.T, class, metric string) {
 }
 
 func TestAutoscaleSustaining(t *testing.T) {
-	// When traffic increases, a knative app should scale up and sustain the scale
-	// as long as the traffic sustains, despite whether it is switching modes between
-	// normal and panic.
 	t.Parallel()
-
-	ctx := SetupSvc(t, autoscaling.KPA, autoscaling.Concurrency, containerConcurrency, targetUtilization)
-	test.EnsureTearDown(t, ctx.Clients(), ctx.Names())
-
-	AssertAutoscaleUpToNumPods(ctx, 1, 10, time.After(2*time.Minute), false /* quick */)
+	for _, algo := range []string{
+		autoscaling.MetricAggregationAlgorithmLinear,
+		autoscaling.MetricAggregationAlgorithmWeightedExponential,
+	} {
+		t.Run("aggregation-"+algo, func(t *testing.T) {
+			// When traffic increases, a knative app should scale up and sustain the scale
+			// as long as the traffic sustains, despite whether it is switching modes between
+			// normal and panic.
+			t.Parallel()
```

Comment:
Is this safe without doing `algo := algo` above the `t.Run`? (I think if Run runs in parallel the loop var can get overwritten before we use it)

Reply:
I actually don't think running those in parallel is good at all